code/modules/projectiles/projectile/bullets/special.dm

```diff
@@ -38,3 +38,176 @@
+	/// When this coin is targeted with a valid splitshot, it creates this many child splitshots
+	var/num_of_splitshots = 2
+	/// The crosshair icon put on the targeted turf for the user- so we can remove it from their images when done
+	var/image/crosshair_indicator
+	/// The mob who originally flipped this coin, as a matter of convenience, may be able tto be removed
+	var/mob/original_firer
+
+/obj/projectile/bullet/coin/Initialize(mapload, turf/the_target, mob/original_firer)
+	src.original_firer = original_firer
+	target_turf = the_target
+	range = (get_dist(original_firer, target_turf) + 3) * 3 // 3 tiles past the origin (the *3 is because Range() ticks 3 times a tile because of the slower speed)
+
+	. = ..()
+
+	if(!istype(original_firer) || !original_firer.client)
+		return
+
+	var/client/firing_client = original_firer.client
+	crosshair_indicator = image('icons/obj/supplypods_32x32.dmi', target_turf, "LZ")
+	firing_client.images += crosshair_indicator
+
+/obj/projectile/bullet/coin/Destroy()
+	remove_crosshair_indicator()
+	return ..()
+
+// the coin must be on the target turf to be directly targetable
+/obj/projectile/bullet/coin/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if(!valid && get_dist(loc, target_turf) < 1)
+		original_firer?.playsound_local(src, 'sound/machines/ping.ogg', 30)
+		valid = TRUE
+	else if(valid && get_dist(loc, target_turf) > 1)
+		valid = FALSE
+		remove_crosshair_indicator()
+
+/// Remove the crosshair indicator from the original firer if it exists
+/obj/projectile/bullet/coin/proc/remove_crosshair_indicator()
+	if(original_firer?.client && crosshair_indicator)
+		original_firer.client.images -= crosshair_indicator
+	QDEL_NULL(crosshair_indicator)
+
+/// We've been shot by a marksman revolver shot, or the ricochet off another coin, check if we can actually ricochet. The forced var being TRUE means it's a ricochet from another coin
+/obj/projectile/bullet/coin/proc/check_splitshot(mob/living/shooter, obj/projectile/bullet/marksman/incoming_shot, forced = FALSE)
+	if(!forced && get_dist(src, target_turf) > 1)
+		return FALSE
+
+	used = TRUE
+	var/turf/cur_tur = get_turf(src)
+	cur_tur.visible_message(span_nicegreen("[incoming_shot] impacts [src] and splits!"))
+	iterate_splitshots(shooter, incoming_shot)
+	QDEL_IN(src, 0.25 SECONDS) // may not be needed
+
+/// Now we actually create all the splitshots, loop through however many splits we'll create and fire them
+/obj/projectile/bullet/coin/proc/iterate_splitshots(mob/living/shooter, obj/projectile/incoming_shot)
+	for(var/i in 1 to num_of_splitshots)
+		fire_splitshot(incoming_shot)
+
+/// Minor convenience function for creating each shrapnel piece with circle explosions, mostly stolen from the MIRV component
+/obj/projectile/bullet/coin/proc/fire_splitshot(obj/projectile/bullet/marksman/incoming_shot)
+	var/atom/next_target = find_next_target()
+
+	ADD_TRAIT(next_target, TRAIT_RECENTLY_COINED, ABSTRACT_ITEM_TRAIT)
```

Comment:
this isnt an abstract item

Reply:
Yeah just replace this with `"[src]"`, we should be more judicious about where we use explicit trait sources anyway